app/Keycloak/Dto/Client.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Keycloak\Dto;
+
+final class Client
+{
+    private string $id;
+    private string $clientId;
+    private string $name;
+    private string $description;
+    private ?string $rootUrl;
+    private ?string $adminUrl;
+    private ?string $baseUrl;
+    private bool $surrogateAuthRequired;
+    private bool $enabled;
+    private bool $alwaysDisplayInConsole;
+    private string $clientAuthenticatorType;
+    private ?string $secret;
+    private array $redirectUris;
+    private array $webOrigins;
+    private int $notBefore;
+    private bool $bearerOnly;
+    private bool $consentRequired;
+    private bool $standardFlowEnabled;
+    private bool $implicitFlowEnabled;
+    private bool $directAccessGrantsEnabled;
+    private bool $serviceAccountsEnabled;
+    private bool $publicClient;
+    private bool $frontchannelLogout;
+    private string $protocol;
+    private array $attributes;
+    private array $authenticationFlowBindingOverrides;
+    private bool $fullScopeAllowed;
+    private int $nodeReRegistrationTimeout;
+    private array $protocolMappers;
+    private array $defaultClientScopes;
+    private array $optionalClientScopes;
+    private array $access;
```

Comment:
Maybe these properties can be grouped more into other classes. e.g. scopes, urls, flags

Reply:
I also wonder if we need all these values in the context of publiq-platform? If we take the scope of this feature into account the final solution needs to create clients inside all the configured Keycloack environments like we do for Auth0 and UiTiD. We don't need to know all the details, just the client id and the secret.

Reply:
I know you work step by step, which is great, but I think we miss an essential part and that is how we will store the created clients inside publiq-platform. Because those credentials are stored internally.